web_ui/src/shared/components/media-items-list/use-scroll-to-target-item.hook.tsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2022-2025 Intel Corporation
+// LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
+
+import { DependencyList, useEffect } from 'react';
+
+import { isNil } from 'lodash-es';
+
+interface useScrollToTargetItemProps {
+    gap: number;
+    dependencies?: DependencyList;
+    container?: Element | null;
+    targetIndex?: number;
+    callback: (scrollTo: number) => void;
+}
+
+const isValidIndex = (index?: number): index is number => !isNil(index) && Number.isInteger(index) && index >= 0;
+
+export const useScrollToTargetItem = ({
+    gap,
+    container,
+    targetIndex,
+    dependencies = [],
+    callback,
+}: useScrollToTargetItemProps) => {
+    useEffect(() => {
+        const timeoutId = setTimeout(() => {
+            if (!container || !isValidIndex(targetIndex)) {
+                return;
+            }
+
+            const containerWidth = container.clientWidth;
+            const childrenWidth = container.firstElementChild?.clientWidth ?? 1;
+            const childrenHeight = container.firstElementChild?.clientHeight ?? 1;
+            const childrenPreRow = Math.floor(containerWidth / childrenWidth);
+            const targetRow = Math.floor(targetIndex / childrenPreRow);
+            const scrollTo = (childrenHeight + gap) * targetRow;
+
+            callback(scrollTo);
+            // we don't want to scroll immediately
+            // in case of changed view mode we have to scroll once view is rendered
+        }, 500);
```

Comment:
a slow network might break this behavior. Could we not handle this on the parent? Maybe pass a "isReady" or "isEnabled" prop once the parent renders, perhaps inside a useEffect?

Reply:
The hook re-executes when `scrollToIndex` changes. Once the items are loaded and the index is set, it responds as expected. I tested it on a 3G connection, and it worked fine